server/models/project.js

```diff
@@ -4,9 +4,25 @@ import slugify from 'slugify';
 
 const { Schema } = mongoose;
 
+const MAX_PROJECT_NAME_LENGTH = 256;
+const MAX_FILE_NAME_LENGTH = 256;
+
+const fileNameValidator = (file) => {
+  const fileNameWithoutExtension = file.replace(/\.[^/.]+$/, ''); // Removes the extension from the fileName
+  if (fileNameWithoutExtension.length === 0) {
+    return false;
+  }
+
+  return true;
+};
+
 const fileSchema = new Schema(
   {
-    name: { type: String, default: 'sketch.js' },
+    name: {
```

Comment:
if the pre 'save' function is running and truncating the file name, then why is a fileNameValidator also necessary?

Reply:
@catarak That's a precaution. Let's say we change the update function later, this should not break right.

Reply:
my concern is that this creates an error that is not being handled, so if you're going to keep it, i'd recommend also checking for this error (maybe something like https://mongoosejs.com/docs/validation.html#custom-validators). otherwise, i think it makes sense to take it out. also, if you're going to add this validator for the file name length, why not for a sketch name length as well?

Reply:
@catarak well, this custom validator exists just for filename because there exist inbuilt min and max length validator for sketch name ( which I thought I added it but realised just now that it isn't there ). Why a custom validator for filename is that the length validation must be done on it **without considering the extension**.<br>
I don't clearly get the error that's not being handled( The validator returns an error telling that "filename validation failed" in the case of failure ). <br>
So either of the following can be done:
. remove the validation as you said earlier.
. add min and max validations for sketchname and min validation for filename

Reply:
the validator does return an error; however, the error is not actually caught if it is thrown. instead, saving the file would just fail.